index.js

```diff
@@ -91,6 +100,28 @@ function _insertCredentials(username, userhash) {
 let cachedAgents;
 
 /**
+ * Generates and maps an internal username to the given username
+ *
+ * To protect against username collisions, this function takes
```

Comment:
I actually intentionally left that part out, but I can put it in if we want. I left it out because this not only protects against test parallelization, but also if someone screws up and doesn't properly clean up after their test, it doesn't break other people's tests.